<a href="https://colab.research.google.com/github/maryum316/Detect-Distance-D435i/blob/main/Copy_of_GEOL117_Module1_Lab.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Lab 1: Introduction to Python, Google CoLab, and Jupyter Notebook

Welcome to Lab 1! This lab is designed to help you familiarize yourself with Python, Google CoLab, Jupyter Notebook, and GitHub. By the end of this lab, you should be able to execute code in a Jupyter Notebook, make simple modifications to the code, and copy and paste a section of code to start a modified section.

#Part 1: Introduction to Python

Python is a popular programming language that is widely used in the scientific and data analysis communities. It is known for its simplicity, readability, and flexibility.

In this section, you will learn how to execute basic Python code in a Jupyter Notebook.

First, let's try running a simple print statement. In the cell below, type print("Hello, World!") and press Shift+Enter to run the code.


In [ ]:
from IPython.display import HTML

HTML('<iframe id="kmsembed-1_vcoh14q1" width="640" height="394" src="https://mediaspace.illinois.edu/embed/secure/iframe/entryId/1_vcoh14q1/uiConfId/26883701/st/0" class="kmsembed" allowfullscreen webkitallowfullscreen mozAllowFullScreen allow="autoplay *; fullscreen *; encrypted-media *" referrerPolicy="no-referrer-when-downgrade" sandbox="allow-downloads allow-forms allow-same-origin allow-scripts allow-top-navigation allow-pointer-lock allow-popups allow-modals allow-orientation-lock allow-popups-to-escape-sandbox allow-presentation allow-top-navigation-by-user-activation" frameborder="0" title="GEOL117_L1_Vid1"></iframe>')


/usr/local/lib/python3.10/dist-packages/IPython/core/display.py:724: UserWarning: Consider using IPython.display.IFrame instead
  warnings.warn("Consider using IPython.display.IFrame instead")


In [ ]:
print("Hello, World!")

Hello, World!



Great! You just ran your first Python code.

Now let's try a simple mathematical operation. In the cell below, type 2 + 2 and press Shift+Enter to run the code.

In [ ]:
2 + 2

4

### Variables and Data Types

In Python, you can create variables to store values. A variable is a named location in memory where you can store a value.

To create a variable in Python, you can use the `=` operator. For example:


In [ ]:
x = 10
y = 20
z = x + y

# Print the value of z
print(z)

30


This code creates three variables: x, y, and z. The value of x is 10, the value of y is 20, and the value of z is the sum of x and y, which is 30.

In Python, variables can be of different data types, such as integers, floats, strings, and booleans.

An integer is a whole number, such as 10, 20, or 30.

A float is a decimal number, such as 3.14, 2.718, or 1.6180.

A string is a sequence of characters, such as "Hello", "World", or "Python". Strings must be enclosed in quotation marks.

A boolean is a value that is either True or False.

You can use the type function to determine the data type of a variable. For example:

In [ ]:
# Create variables x, y, and z
x = 10
y = 20.0
z = "Python"

# Print the data type of each variable
print(type(x))
print(type(y))
print(type(z))

<class 'int'>
<class 'float'>
<class 'str'>


This code creates three variables: x, y, and z. The value of x is an integer, the value of y is a float, and the value of z is a string.

Next, here is an example of calculating the area of a circle in Python:

In [ ]:
# Import the math module
import math

# Define the radius of the circle
radius = 5

# Calculate the area of the circle
area = math.pi * radius**2

# Print the result
print(area)

78.53981633974483


This code calculates the area of a circle with radius 5, which is approximately 78.54.

# Basic Operations
In Python, you can perform basic arithmetic operations such as addition, subtraction, multiplication, and division.

For example:

In [ ]:
# Add two numbers
x = 10
y = 20
z = x + y
print(z)  # prints 30

# Subtract two numbers
x = 10
y = 20
z = x - y
print(z)  # prints -10

# Multiply two numbers
x = 10
y = 20
z = x * y
print(z)  # prints 200

# Divide two numbers
x = 10
y = 20
z = x / y
print(z)  # prints 0.5

30
-10
200
0.5


You can also use the % operator to calculate the remainder of a division. For example:

In [ ]:
# Calculate the remainder of a division
x = 10
y = 3
z = x % y
print(z)  # prints 1

1


You can use the ** operator to calculate the power of a number. For example:

In [ ]:
# Calculate the power of a number
x = 2
y = 3
z = x ** y
print(z)  # prints 8

8


In [ ]:
from IPython.display import HTML

HTML('<iframe id="kmsembed-1_eqqza7cv" width="640" height="394" src="https://mediaspace.illinois.edu/embed/secure/iframe/entryId/1_eqqza7cv/uiConfId/26883701/st/0" class="kmsembed" allowfullscreen webkitallowfullscreen mozAllowFullScreen allow="autoplay *; fullscreen *; encrypted-media *" referrerPolicy="no-referrer-when-downgrade" sandbox="allow-downloads allow-forms allow-same-origin allow-scripts allow-top-navigation allow-pointer-lock allow-popups allow-modals allow-orientation-lock allow-popups-to-escape-sandbox allow-presentation allow-top-navigation-by-user-activation" frameborder="0" title="GEOL117_L1_Vid2"></iframe>')


# Lists
In Python, you can create a list to store a collection of values. A list is an ordered collection of items, and each item can be of a different data type.

To create a list, you can use square brackets [] and separate the items with commas. For example:



In [ ]:
# Create a list
my_list = ["apple", "banana", "cherry"]

# Access the first item
print(my_list[0])  # prints "apple"

# Access the second item
print(my_list[1])  # prints "banana"

# Access the third item
print(my_list[2])  # prints "cherry"


apple
banana
cherry


You can modify the items in a list by assigning a new value to them. For example:

In [ ]:
# Create a list
my_list = ["apple", "banana", "cherry"]

# Modify the second item
my_list[1] = "orange"
print(my_list)  # prints ["apple", "orange", "cherry"]


['apple', 'orange', 'cherry']


You can add items to a list using the append method. For example:

In [ ]:
# Create a list
my_list = ["apple", "banana", "cherry"]

# Add a new item to the list
my_list.append("orange")
print(my_list)  # prints ["apple", "banana", "cherry", "orange"]


['apple', 'banana', 'cherry', 'orange']


You can remove items from a list using the remove method. For example:

In [ ]:
# Create a list
my_list = ["apple", "banana", "cherry"]

# Remove the second item from the list
my_list.remove("banana")
print(my_list)  # prints ["apple", "cherry"]


['apple', 'cherry']


You can also use the del statement to remove items from a list. For example:



In [ ]:
# Create a list
my_list = ["apple", "banana", "cherry"]

# Remove the second item from the list
del my_list[1]
print(my_list)  # prints ["apple", "cherry"]


['apple', 'cherry']


In [ ]:
from IPython.display import HTML

HTML('<iframe id="kmsembed-1_otxcggwa" width="640" height="394" src="https://mediaspace.illinois.edu/embed/secure/iframe/entryId/1_otxcggwa/uiConfId/26883701/st/0" class="kmsembed" allowfullscreen webkitallowfullscreen mozAllowFullScreen allow="autoplay *; fullscreen *; encrypted-media *" referrerPolicy="no-referrer-when-downgrade" sandbox="allow-downloads allow-forms allow-same-origin allow-scripts allow-top-navigation allow-pointer-lock allow-popups allow-modals allow-orientation-lock allow-popups-to-escape-sandbox allow-presentation allow-top-navigation-by-user-activation" frameborder="0" title="GEOL117_L1_Vid3"></iframe>')


# Tuples
In Python, you can create a tuple to store a collection of values. A tuple is similar to a list, but it is immutable, which means that you cannot modify the items in a tuple.

To create a tuple, you can use parentheses () and separate the items with commas. For example:



In [ ]:
# Create a tuple of integers
my_tuple = (10, 20, 30, 40, 50)

# Create a tuple of strings
my_tuple = ("apple", "banana", "cherry")

# Create a tuple of mixed data types
my_tuple = (10, "apple", 3.14, True)


You can access the items in a tuple using their indices. The indices start at 0, so the first item has index 0, the second item has index 1, and so on.

For example:

In [ ]:
# Create a tuple
my_tuple = ("apple", "banana", "cherry")

# Access the first item
print(my_tuple[0])  # prints "apple"

# Access the second item
print(my_tuple[1])  # prints "banana"

# Access the third item
print(my_tuple[2])  # prints "cherry"


apple
banana
cherry


You cannot modify the items in a tuple. If you try to assign a new value to an item in a tuple, you will get a TypeError message.

For example:

In [ ]:
# Create a tuple
my_tuple = ("apple", "banana", "cherry")

# Try to modify the second item
my_tuple[1] = "orange"  # This line will cause a TypeError


TypeError: ignored

In [ ]:
from IPython.display import HTML

HTML('<iframe id="kmsembed-1_ckwe3az5" width="640" height="394" src="https://mediaspace.illinois.edu/embed/secure/iframe/entryId/1_ckwe3az5/uiConfId/26883701/st/0" class="kmsembed" allowfullscreen webkitallowfullscreen mozAllowFullScreen allow="autoplay *; fullscreen *; encrypted-media *" referrerPolicy="no-referrer-when-downgrade" sandbox="allow-downloads allow-forms allow-same-origin allow-scripts allow-top-navigation allow-pointer-lock allow-popups allow-modals allow-orientation-lock allow-popups-to-escape-sandbox allow-presentation allow-top-navigation-by-user-activation" frameborder="0" title="GEOL117_L1_Vid4"></iframe>')


# Dictionaries
In Python, you can create a dictionary to store a collection of key-value pairs. A dictionary is a collection of items that are stored in the form of key-value pairs, where the key is a unique identifier and the value is the data.

To create a dictionary, you can use curly braces {} and specify the key-value pairs using the format key: value. For example:

In [ ]:
# Create a dictionary of integers
my_dict = {
    "a": 10,
    "b": 20,
    "c": 30
}

# Create a dictionary of strings
my_dict = {
    "a": "apple",
    "b": "banana",
    "c": "cherry"
}

# Create a dictionary of mixed data types
my_dict = {
    "a": 10,
    "b": "banana",
    "c": 3.14
}


You can access the values in a dictionary using the keys. For example:


In [ ]:
# Create a dictionary
my_dict = {
    "a": "apple",
    "b": "banana",
    "c": "cherry"
}

# Access the value for the key "a"
print(my_dict["a"])  # prints "apple"

# Access the value for the key "b"
print(my_dict["b"])  # prints "banana"

# Access the value for the key "c"
print(my_dict["c"])  # prints "cherry"


apple
banana
cherry


You can modify the values in a dictionary by assigning a new value to a key. For example:



In [ ]:
# Create a dictionary
my_dict = {
    "a": "apple",
    "b": "banana",
    "c": "cherry"
}

# Modify the value for the key "a"
my_dict["a"] = "orange"
print(my_dict)  # prints { "a": "orange", "b": "banana", "c": "cherry" }


{'a': 'orange', 'b': 'banana', 'c': 'cherry'}


You can add new key-value pairs to a dictionary using the assignment operator =. For example:



In [ ]:
# Create a dictionary
my_dict = {
    "a": "apple",
    "b": "banana",
    "c": "cherry"
}

# Add a new key-value pair to the dictionary
my_dict["d"] = "orange"
print(my_dict)  # prints { "a": "apple", "b": "banana", "c": "cherry", "d": "orange" }


{'a': 'apple', 'b': 'banana', 'c': 'cherry', 'd': 'orange'}


You can remove key-value pairs from a dictionary using the del statement. For example:



In [ ]:
# Create a dictionary
my_dict = {
    "a": "apple",
    "b": "banana",
    "c": "cherry"
}

# Remove the key-value pair for the key "b"
del my_dict["b"]
print(my_dict)  # prints { "a": "apple", "c": "cherry" }


{'a': 'apple', 'c': 'cherry'}


In [ ]:
from IPython.display import HTML

HTML('<iframe id="kmsembed-1_c3wsuqvq" width="640" height="394" src="https://mediaspace.illinois.edu/embed/secure/iframe/entryId/1_c3wsuqvq/uiConfId/26883701/st/0" class="kmsembed" allowfullscreen webkitallowfullscreen mozAllowFullScreen allow="autoplay *; fullscreen *; encrypted-media *" referrerPolicy="no-referrer-when-downgrade" sandbox="allow-downloads allow-forms allow-same-origin allow-scripts allow-top-navigation allow-pointer-lock allow-popups allow-modals allow-orientation-lock allow-popups-to-escape-sandbox allow-presentation allow-top-navigation-by-user-activation" frameborder="0" title="GEOL117_L1_Vid5"></iframe>')


# Loops
In Python, you can use loops to repeat a block of code multiple times.

**'for' loops**

A 'for' loop allows you to iterate over a sequence of items, such as a list or a string.

For example:

In [ ]:
# Print the numbers from 1 to 5
for i in range(1, 6):
    print(i)


1
2
3
4
5


The range function generates a sequence of numbers from the start value to the end value (not inclusive).

You can also use a for loop to iterate over a list of items.

For example:

In [ ]:
# Create a list of fruits
fruits = ["apple", "banana", "cherry"]

# Print each fruit in the list
for fruit in fruits:
    print(fruit)


apple
banana
cherry


You can use the enumerate function to iterate over a list of items and get the index of each item.

For example:

In [ ]:
# Create a list of fruits
fruits = ["apple", "banana", "cherry"]

# Print each fruit in the list
for fruit in fruits:
    print(fruit)


apple
banana
cherry


**'while' loops**
A 'while' loop allows you to repeat a block of code as long as a certain condition is met.

For example:

In [ ]:
# Print the numbers from 1 to 5
i = 1
while i <= 5:
    print(i)
    i += 1


1
2
3
4
5


You can use the 'break' statement to exit a loop early.

For example:

In [ ]:
# Print the numbers from 1 to 5, but exit the loop early if the number is 3
i = 1
while i <= 5:
    if i == 3:
        break
    print(i)
    i += 1


1
2


You can use the 'continue' statement to skip the rest of the current iteration and move on to the next one.

For example:

In [ ]:
# Print the numbers from 1 to 5, but skip the number 3
i = 1
while i <= 5:
    if i == 3:
        i += 1
        continue
    print(i)
    i += 1


1
2
4
5


#Functions
In Python, you can define a function to encapsulate a block of code that you want to reuse.

To define a function, you can use the def keyword followed by the function name and a set of parentheses () that may contain parameters. The code block for the function is indented and starts with a colon :.

For example:

In [ ]:
def greet(name):
    """
    Prints a greeting for the given name.
    """
    print(f"Hello, {name}!")


To call a function, you can use the function name followed by a set of parentheses () that may contain arguments.

For example:

In [ ]:
# Define a function
def greet(name):
    """
    Prints a greeting for the given name.
    """
    print(f"Hello, {name}!")

# Call the function
greet("John")  # prints "Hello, John!"
greet("Mary")  # prints "Hello, Mary!"


Hello, John!
Hello, Mary!


You can define a function with parameters to accept input values. The parameters are specified in the parentheses () after the function name.

For example:

In [ ]:
# Define a function with parameters
def add(x, y):
    """
    Returns the sum of x and y.
    """
    return x + y

# Call the function with arguments
result = add(10, 20)  # result is 30


You can define a function with a default value for a parameter. If the caller does not provide a value for the parameter, the default value will be used.

For example:

In [ ]:
# Define a function with a default value for a parameter
def add(x, y=10):
    """
    Returns the sum of x and y.
    """
    return x + y

# Call the function with one argument
result = add(10)  # result is 20

# Call the function with two arguments
result = add(10, 20)  # result is 30


You can define a function with a variable number of arguments by using the *args parameter. The *args parameter allows the function to accept any number of arguments as a tuple.

For example:

In [ ]:
# Define a function with a variable number of arguments
def add(*args):
    """
    Returns the sum of all arguments.
    """
    total = 0
    for i in args:
        total += i
    return total

# Call the function with multiple arguments
result = add(1, 2, 3, 4, 5)  # result is 15


You can define a function with a variable number of keyword arguments by using the **kwargs parameter. The **kwargs parameter allows the function to accept any number of keyword arguments as a dictionary.

For example:

In [ ]:
# Define a function with a variable number of keyword arguments
def add(**kwargs):
    """
    Returns the sum of all keyword arguments.
    """
    total = 0
    for i in kwargs.values():
        total += i
    return total

# Call the function with multiple keyword arguments
result = add(x=1, y=1)


As you can see, there are many basic concepts in Python that you can use to write simple programs. However, this is just the tip of the iceberg! There are many more advanced features and libraries available in Python that you can use to write more complex programs. While it may be tempting to try to cover everything in one lab, it's important to remember that programming takes time and practice. In this lab, you have learned some basic concepts and had a chance to try them out. In the following weeks, you will have more opportunities to build upon this foundation and learn new skills. So don't worry if you don't feel like an expert yet, programming is a journey and you will continue to improve with practice.

# Part 2: Introduction to Google CoLab and Jupyter Notebook
In this section, you will learn how to use Google CoLab and Jupyter Notebook to run Python code.

Google CoLab is a cloud-based programming environment that allows you to write and execute code from anywhere with an internet connection. It is particularly useful for data analysis and machine learning tasks, as it allows you to access powerful computing resources without the need to set up a local programming environment.

Jupyter Notebook is a web-based application that allows you to create and share documents that contain live code, equations, visualizations, and explanatory text. It is a popular tool for data scientists and is often used in conjunction with Google CoLab.

To get started with Google CoLab and Jupyter Notebook, you will need to link your GitHub account with Google CoLab. Follow these steps to do so:


1.   Go to https://colab.research.google.com/github/. You may need to adjust your pop-up settings in your browser.

2.   If you are not already logged in to GitHub, you will be prompted to do so.  If you are already logged in, you will be asked to authorize Colaboratory.

3.   You should see a pop-up window with an orange header. Make sure that the option to "Include private repos" is checked.

4. Navigate to your desired repository and click on the ".ipynb" file you want to work with. Make sure that your user name is displayed, not someone else's.

5. The Jupyter Notebook will open in a new tab in your browser.

Now that you have a Jupyter Notebook open, let's try running some code. In the cell below, type print("Hello, CoLab!") and press Shift+Enter to run the code.

In [ ]:
print("Hello, CoLab!")

Hello, CoLab!


You can also use Markdown cells to write explanations, equations, and other text in your Jupyter Notebook. To create a Markdown cell, click on the cell and select "Markdown" from the dropdown menu in the toolbar. Then, you can type your text in the cell and press Shift+Enter to render it.

For example, to write an equation in a Markdown cell, you can use LaTeX syntax. LaTeX is a typesetting system that is widely used in the scientific community for formatting equations and other technical documents.

To write an equation in LaTeX, enclose it in dollar signs ($). For example, to write the equation for the area of a circle, you can use the following code:

In [1]:
from math import pi
r = float(input("Enter the radius of the circle: "))
A = pi * r**2
print(f"The area of the circle with radius {r} is {A:.2f}")

Enter the radius of the circle: 4
The area of the circle with radius 4.0 is 50.27


Try writing these equations in a Markdown cell below.

# Part 3: Modifying and Copying Code

Now that you have learned the basics of Python and Jupyter Notebook, let's try making some simple modifications to the code.

In the cell below, write a for loop that prints the numbers from 1 to 10. You can use the following code as a starting point:

In [ ]:
for i in range(1, 11):
  print(i)

1
2
3
4
5
6
7
8
9
10


Try modifying the code to print the numbers from 1 to 20 instead.

In [ ]:
for i in range(1, 21):
  print(i)

1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20


Great! Now let's try copying and pasting a section of code.

In the cell below, copy and paste the code you just wrote, but modify it to print the squares of the numbers from 1 to 10 instead of the numbers themselves.

In [ ]:
for i in range(1, 11):
  print(i**2)

1
4
9
16
25
36
49
64
81
100


## Part 4: Submitting Your Work

### Using Google Colab's GitHub Integration with GitHub Classroom

To easily submit your work for this lab via GitHub Classroom and Google Colab's GitHub Integration, follow these steps:

1. **Save Your Work in Colab**:
   - Before proceeding, ensure you've saved the latest changes in your Colab notebook.
   - You can do this using the `File > Save` menu or by pressing the `Ctrl + S` keyboard shortcut.

2. **Link Colab to GitHub**:
   - Navigate to the `File` option located at the top left corner of your Colab notebook.
   - Select `Save a copy in GitHub` from the dropdown.
   - If you're linking Colab with GitHub for the first time, you'll be prompted for permission. Please grant the necessary permissions.

3. **Commit to Your GitHub Classroom Repository**:
   - A dialog box will appear after step 2.
   - Choose your specific repository for this assignment from the dropdown (it should be automatically created via GitHub Classroom).
   - Type in a meaningful commit message. For this lab, something like "Completed Lab 1" would be appropriate.
   - Click `OK`.

4. **Verify Your Commit on GitHub**:
   - Open your web browser and visit your GitHub repository for this assignment.
   - Confirm that your latest changes are visible and that the commit message you provided appears in the commit history.

That's it! You have successfully committed your assignment via GitHub Classroom. Your instructor will be able to access and grade your submission directly through GitHub.
